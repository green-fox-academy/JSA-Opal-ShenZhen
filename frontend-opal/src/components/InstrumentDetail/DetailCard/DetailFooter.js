import React from 'react';
import { Content, Text, CardItem, Icon, Right } from 'native-base';

import styles from '../styles';

const DetailFooter = () => {
  return (
    <CardItem style={styles.DetailFooter}>
      <Icon name="heart" />
      <Icon name="share" />
      <Icon name="bookmark" />
    </CardItem>
  );
};

export default DetailFooter;
