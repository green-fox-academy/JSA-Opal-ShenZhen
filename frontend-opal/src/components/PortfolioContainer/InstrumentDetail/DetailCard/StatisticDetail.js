import React from 'react';
import { View } from 'react-native';
import { Container, CardItem, Body, Text } from 'native-base';

import styles from '../styles';

const StatisticDetail = ({ data }) => {
  const renderItems = items =>
    items.map(item => (
      <View key={item.id} style={styles.detailItem}>
        <Text style={styles.itemText}>{item.name}:</Text>
        <Text style={styles.itemText}>{item.value}</Text>
      </View>
    ));

  return <View>{renderItems(data)}</View>;
};

export default StatisticDetail;
