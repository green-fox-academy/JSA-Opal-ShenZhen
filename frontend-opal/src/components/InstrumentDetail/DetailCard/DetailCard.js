import React from 'react';
import { View } from 'react-native';
import { Content, Card, Text, Spinner } from 'native-base';

import presetProps from 'components/InstrumentDetail/presetProps';
import StatisticDetail from './StatisticDetail';
import DetailFooter from './DetailFooter';
import styles from '../styles';

const DetailCard = ({ apiData }) => {
  const getDataGroup = (data, groupNum) => data.filter(item => item.group === groupNum);

  return (
    <View>
      <Content padder style={styles.secondCard}>
        <Card transparent>
          <Text style={styles.title}>Details</Text>
        </Card>

        <Card>
          <View style={styles.detail}>
            {apiData.length === 0 ? (
              <View style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
                <Spinner color="#e17055" />
              </View>
            ) : (
              <>
                <StatisticDetail data={getDataGroup(apiData, 1)} />
                <StatisticDetail data={getDataGroup(apiData, 2)} />
              </>
            )}
          </View>

          <DetailFooter />
        </Card>
      </Content>
    </View>
  );
};

DetailCard.defaultProps = {
  apiData: []
};

DetailCard.propTypes = {
  apiData: presetProps.detailDataTypes
};

export default DetailCard;
