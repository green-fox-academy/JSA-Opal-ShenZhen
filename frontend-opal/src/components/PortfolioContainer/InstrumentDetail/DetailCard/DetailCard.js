import React from 'react';
import { View } from 'react-native';
import { Content, Container, Card, CardItem, Text } from 'native-base';

import StatisticDetail from './StatisticDetail';
import DetailFooter from './DetailFooter';
import styles from '../styles';

import dataSample from '../dataSample';

const DetailCard = () => {
  const getDataGroup = (data, groupNum) => data.filter(item => item.group === groupNum);

  return (
    <Content padder style={styles.secondCard}>
      <Card transparent>
        <Text style={styles.title}>Details</Text>
      </Card>

      <Card>
        <View style={styles.detail}>
          <StatisticDetail data={getDataGroup(dataSample, 1)} />
          <StatisticDetail data={getDataGroup(dataSample, 2)} />
        </View>

        <DetailFooter />
      </Card>
    </Content>
  );
};

export default DetailCard;
