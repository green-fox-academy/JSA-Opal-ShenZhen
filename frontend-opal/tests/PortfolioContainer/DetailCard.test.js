import React from 'react';
import renderer from 'react-test-renderer';
import DetailCard from 'components/PortfolioContainer/InstrumentDetail/DetailCard';

describe('DetailCard component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DetailCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
