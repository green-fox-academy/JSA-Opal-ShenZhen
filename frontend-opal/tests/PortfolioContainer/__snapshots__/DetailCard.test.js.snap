// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailCard component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 2,
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomWidth": 0,
          "borderColor": "#ccc",
          "borderLeftWidth": 0,
          "borderRadius": 2,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "elevation": null,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      transparent={true}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontSize": 24,
            },
          ]
        }
        uppercase={false}
      >
        Details
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#DFF1FD",
            "display": "flex",
            "flexDirection": "row",
            "height": 200,
            "justifyContent": "space-evenly",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#e17055"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "height": 80,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "display": "flex",
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <Text />
        <Text />
        <Text />
      </View>
    </View>
  </View>
</View>
`;
